### PR TITLE
feat: Allow shared calendars as appointment conflict calendars

### PR DESCRIPTION
**Pull request by SebastianKrupinski**
Resolves: https://github.com/nextcloud/calendar/issues/3786
Requires: https://github.com/nextcloud/server/pull/48621

## Summary
Adjusted "Additional calendars to check for conflicts" selection logic to list all users calendars, personal and shared to be selectable for conflicts.

### CODE REVIEW

**Review by ChristophWurst — COMMENTED** [review]
The app also works with 30, so a conditional logic for 30 vs 31 is necessary

Ref https://github.com/nextcloud/calendar/blob/562e828182d041bfe25a5052679920119fc7f455/appinfo/info.xml#L43

**Review by miaulalala — APPROVED** [review]

**Comment by SebastianKrupinski** [comment]
> The app also works with 30, so a conditional logic for 30 vs 31 is necessary
> 
> Ref
> 
> https://github.com/nextcloud/calendar/blob/562e828182d041bfe25a5052679920119fc7f455/appinfo/info.xml#L43

@ChristophWurst Done. Please have another look.

File: src/components/AppointmentConfigModal.vue [hunk 1]
```diff
@@ -216,26 +217,33 @@ export default {
 		},
 		calendar() {
 			if (!this.editing.targetCalendarUri) {
-				return this.ownSortedCalendars[0]
+				return this.sortedCalendars[0]
```
Comment (st3iny) [inline src/components/AppointmentConfigModal.vue:220]:
```suggestion
				return this.availableCalendars[0]
```

This should also be affected by the feature gate, right?

File: src/components/AppointmentConfigModal.vue [hunk 2]
```diff
@@ -216,26 +217,33 @@ export default {
...
 			return freebusyUris.map(uri => {
-				return this.ownSortedCalendars.find(cal => this.calendarUrlToUri(cal.url) === uri)
-			})
+				return this.availableCalendars.find(cal => this.calendarUrlToUri(cal.url) === uri)
+			}).filter(calendar => calendar !== undefined)
 		},
 		defaultConfig() {
 			return AppointmentConfig.createDefault(
-				this.calendarUrlToUri(this.ownSortedCalendars[0].url),
+				this.calendarUrlToUri(this.sortedCalendars[0].url),
```
Comment (st3iny) [inline src/components/AppointmentConfigModal.vue:246]:
```suggestion
				this.calendarUrlToUri(this.availableCalendars[0].url),
```

See above.

**Review by st3iny — COMMENTED** [review]
Looks good otherwise.

File: src/components/AppointmentConfigModal.vue (hunk 2 above)
Comment (SebastianKrupinski) [reply src/components/AppointmentConfigModal.vue]: Done.

File: src/components/AppointmentConfigModal.vue (hunk 1 above)
Comment (SebastianKrupinski) [reply src/components/AppointmentConfigModal.vue]: Done.

Those do not need to be feature gated. But I changed them to use the feature gate, so the logic is easier to understand. The end result is the same.

**Review by st3iny — APPROVED** [review]
Looks good and works. I reverted the system version initial state changes to expedite the process a bit.

The version of Nextcloud is already available int the frontend via `window.OC.config.version` and doesn't need to be provided via initial state.

**Comment by SebastianKrupinski** [comment]
> The version of Nextcloud is already available in the frontend via `window.OC.config.version` and doesn't need to be provided via initial state.

I wish I knew this before all the changes I made! (face palm)... Learning moment

**Pull request merged**